packages/tx/src/legacyTransaction.ts

```diff
@@ -135,6 +135,16 @@ export default class Transaction extends BaseTransaction<Transaction> {
       }
     }
 
+    if (this.common.isActivatedEIP(3860)) {
```

Comment:
This block is in all three transaction types, the reason is that we do not have access to common in baseTransaction. Adding it in baseTransaction is somewhat hard as especially legacyTransaction has some extra logic to extract chainId.

Reply:
hm. i don't like how we have to keep duplicating code across the txs classes like this. but thanks for the reasoning why.

would be much better if we always write the code just once. what if we move it to a method in`util.ts` called e.g. `checkMaxInitCodeSize()`, then we can just keep the if isActivatedEIP(3860) and call the method inside.

Reply:
other than that, PR looks great! I'm not sure why the errors don't have any coverage on them though (I'm seeing codecov warnings), i assume the added TransactionTests should cover that. if not maybe we should add a few explicit tests for coverage.

Reply:
Yep, great suggestion moving it to `util`! I also did not like it.

Reply:
Addressed this